python/ray/data/datasource/parquet_datasource.py

```diff
@@ -56,6 +56,61 @@ def _deregister_parquet_file_fragment_serialization():
     ray.util.deregister_serializer(ParquetFileFragment)
 
 
+# this retry helps when HA hdfs service not able to handle overloaded read request.
+# even with HA configed, hdfs service might be overloaded with parquet file read
+# request expecially when simutaneously running many hyper parameter tuning jobs
```

Comment:
For your particular use case, IIRC you tried to run some PyArrow processes and make a lot of connections to a HDFS service, but you didn't reproduce the connection issue. Was this not the case any more?

Reply:
Yes, we tried simulate the situation with a lot PyArrow processes along, but not able to reproduce it under such condition.  I will attribute the connection issue to unknown reason in the comments then.